sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -159,6 +206,75 @@ private bool HasNoReturn(MethodDeclarationSyntax method)
             return semanticModel.GetTypeInfo(method.ReturnType).Type.SpecialType == SpecialType.System_Void;
         }
 
+        private void ExportReturnStatement(ReturnStatementSyntax ret, string functionReturnType)
+        {
+            if (ret.Expression == null)
+            {
+                writer.WriteLine($"return {GetLocation(ret)}");
+            }
+            else
+            {
+                Debug.Assert(functionReturnType != "()", "Returning value in function declared with no return type");
+                var returnedVal = ret.Expression.RemoveParentheses();
+                if (semanticModel.GetTypeInfo(returnedVal).Type == null &&
+                    returnedVal.Kind() == SyntaxKind.SimpleMemberAccessExpression)
+                {
+                    // Special case a returning a method group that will be cast into a func
+                    writer.WriteLine($"%{OpId(ret)} = cbde.unknown : none {GetLocation(ret)} // return method group");
+                    writer.WriteLine($"return %{OpId(ret)} : none {GetLocation(ret)}");
+                    return;
+                }
+
+                var id = ComputeCompatibleId(ret.Expression, functionReturnType);
+                writer.WriteLine($"return %{id} : {functionReturnType} {GetLocation(ret)}");
+            }
+        }
+
+        private void ExportJumpBlock(JumpBlock jb, string functionReturnType)
+        {
+            switch (jb.JumpNode)
+            {
+                case ReturnStatementSyntax ret:
+                    ExportReturnStatement(ret, functionReturnType);
+                    break;
+                case BreakStatementSyntax breakStmt:
+                    writer.WriteLine($"br ^{BlockId(jb.SuccessorBlock)} {GetLocation(breakStmt)} // break");
+                    break;
+                case ContinueStatementSyntax continueStmt:
+                    writer.WriteLine($"br ^{BlockId(jb.SuccessorBlock)} {GetLocation(continueStmt)} // continue");
+                    break;
+                case ThrowStatementSyntax throwStmt:
+                    // TODO : Should we transfert to a catch block if we are inside a try/catch?
```

Comment:
typo in transfert
is there an existing ticket for this (looks like a design question?)

Reply:
I already added the reference to https://github.com/SonarSource/SonarCBDE/issues/111. Would you like something else?
And transfert is not a typo, it's just a french word we used because... well because... well, ok, it's a typo :)

Reply:
it seems github is not smart enough to add the comment here when it's sent via email

the link is good